flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/TaskExecutionState.java

```diff
@@ -160,19 +123,10 @@ public TaskExecutionState(JobID jobID, ExecutionAttemptID executionId,
 	 *                            job this update refers to.
 	 */
 	public Throwable getError(ClassLoader usercodeClassloader) {
-		if (this.serializedError == null) {
+		if (this.throwable == null) {
 			return null;
 		}
-
-		if (this.cachedError == null) {
-			try {
-				cachedError = (Throwable) InstantiationUtil.deserializeObject(this.serializedError, usercodeClassloader);
-			}
-			catch (Exception e) {
-				throw new RuntimeException("Error while deserializing the attached exception", e);
-			}
-		}
-		return this.cachedError;
+		return throwable.getError(usercodeClassloader);
```

Comment:
Could we place the last return in an else branch to make the control flow clearer?